engine/api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessState.scala

```diff
@@ -75,6 +85,30 @@ sealed trait StateStatus {
   def name: String
 }
 
+object StateStatus {
+  //This field keeps all available statuses. Remember!! If you want add new status class you have to add it also here!
+  val availableStatusClasses: Map[String, Class[_ <: StateStatus]] = List(
+    classOf[NotEstablishedStateStatus],
+    classOf[StoppedStateStatus],
+    classOf[DuringDeployStateStatus],
+    classOf[FinishedStateStatus],
+    classOf[RunningStateStatus]
+  )
+    .map(clz => clz.getSimpleName -> clz)
+    .toMap[String, Class[_ <: StateStatus]]
+  
+  def codecToStateStatus(codec: StateStatusCodec): StateStatus =
+    availableStatusClasses
+      .get(codec.clazz)
+      .map(_.getConstructor(classOf[String]).newInstance(codec.value))
+      .getOrElse(SimpleStateStatus.Unknown)
```

Comment:
I see two problems here:
1. I don't think that `getOrElse(SimpleStateStatus.Unknown)` is a good solution for validation handling. why not to return Either with some decoding error in this place? How consumer of this api will now that he/she pass invalid string?
2. During `getConstructor(classOf[String]).newInstance` can be thrown lot of strange errors. Maybe would be better to catch them and throw something more meaningful?

Reply:
Hmm, you'a right. Changed.